### ISSUE TITLE
Stripe payment not possible

### ISSUE THREAD

**Issue by mariobehling** [fossasia/open-event-frontend#8869]
Some users are getting this error:

×TypeError: Cannot read properties of undefined (reading 'confirmPayment')

Test event: https://eventyay.com/e/c93ad54c

![Screenshot from 2023-06-29 19-06-19](https://github.com/fossasia/open-event-frontend/assets/1583873/8be1ef58-1ec3-4827-ac03-3263fc384f8e)

**Comment by mariobehling** [comment]
Error from Console

Failed to load resource: the server responded with a status of 500 ()
state-machine.js:86 n
(anonymous) @ state-machine.js:86
state-machine.js:86 TypeError: Cannot read properties of undefined (reading 'confirmPayment')
    at l.<anonymous> (pending.js:304:1)
    at u (polyfill.js:6473:1)
    at Generator._invoke (polyfill.js:6702:1)
    at e.<computed> [as next] (polyfill.js:6525:1)
    at m (pending.js:10:1)
    at a (pending.js:11:1)
(anonymous) @ state-machine.js:86
state-machine.js:86 Sentry Logger [warn]: Event dropped due to being a duplicate of previously captured event.
(anonymous) @ state-machine.js:86
state-machine.js:86 Sentry Logger [log]: Adding outcome: error:event_processor
state-machine.js:86 Sentry Logger [error]: t: An event processor returned null, will not send event.
    at new t (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22047:67)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22091:282
    at Array.<anonymous> (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:72)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19729:63
    at Array.forEach (<anonymous>)
    at e._executeHandlers (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19729:18)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:161
    at new e (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19730:5)
    at e.then (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:8)
    at e._processEvent (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22091:210)
    at e._captureEvent (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22088:211)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22062:91
    at Array.<anonymous> (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:72)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19729:63
    at Array.forEach (<anonymous>)
    at e._executeHandlers (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19729:18)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:161
    at new e (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19730:5)
    at e.then (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19731:8)
    at e.captureException (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22062:65)
    at e._invokeClient (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19714:19)
    at e.captureException (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19690:120)
    at https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22440:189
    at e.withScope (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:19688:5)
    at console.error (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:22438:24)
    at l.<anonymous> (https://eventyay.com/assets/open-event-frontend-3283d56b780938120b3283a5022be2d6.js:9096:43)
    at u (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:1561:147)
    at Generator._invoke (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:1559:7)
    at e.<computed> [as next] (https://eventyay.com/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:1568:90)
    at m (https://eventyay.com/assets/open-event-frontend-3283d56b780938120b3283a5022be2d6.js:9029:41)
    at a (https://eventyay.com/assets/open-event-frontend-3283d56b780938120b3283a5022be2d6.js:9031:15)
(anonymous) @ state-machine.js:86
state-machine.js:86 TypeError: Cannot read properties of undefined (reading 'confirmPayment')
    at l.<anonymous> (pending.js:304:1)
    at u (polyfill.js:6473:1)
    at Generator._invoke (polyfill.js:6702:1)
    at e.<computed> [as next] (polyfill.js:6525:1)
    at m (pending.js:10:1)
    at a (pending.js:11:1)

**Comment by catInOrbit** [comment]
Hi @mariobehling 

When you click on Stripe it'll have to load a few UI boxes for you to input cards information like this. Only after that can the lifecycle of stripePayment be completed 

this is currently not possible in eventyay due to unknown errors

![Image](https://github.com/fossasia/open-event-frontend/assets/49814026/713f786f-9214-449d-9e68-576820d7aa11)

However at the current moment payment in stripe can still performed in test.eventyay.com
Here's a screen cap of the ticket order flow in action

https://drive.google.com/file/d/1382SpyRJHkcDTSDM91YfIsupA2meN-5A/view?usp=drive_link


![Image](https://github.com/fossasia/open-event-frontend/assets/49814026/8c29a738-fcf3-4ae2-9382-a9efcc7b932f)


![Image](https://github.com/fossasia/open-event-frontend/assets/49814026/982221f7-7880-4a14-845a-370ecea9cd2b)


So it begs the question: Why do  we have 2 different systems, 1 test and 1 production, each behaves in a different way ? Have docker images in eventyay production up-to-date with the test-development one ? And which one is the latest version in this case

**Comment by mariobehling** [comment]
So, problem is sometimes it works, sometimes it does not work. I was able to make a successful transmission. Ten minutes later it no longer worked for me. I am aware that I have to click buttons.

**Comment by mariobehling** [comment]
Sentry Report: [EVENTYAY-FRONTEND-380]

Error: Ember Data Request GET https://api.eventyay.com/v1/events/3171/stripe-authorization returned a 500

(https://sentry.fossasia.org/organizations/eventyay/issues/3912/?referrer=github_integration)

```
Error: Ember Data Request GET https://api.eventyay.com/v1/events/3171/stripe-authorization returned a 500
Payload (application/vnd.api+json)
[object Object]
  at Error: Ember Data Request GET  (/v1/events/3171/stripe-authorization%20returned%20a%20500)
  at n (/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:10723:19)
  at new n (/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:10727:54)
  at n.handleResponse (/assets/vendor-c17586068d6783708cc056cbb8ee814e.js:10797:26)
  at n.handleResponse (/assets/open-event-frontend-a8b63ab74c31d9e91af3b37e1cdfb0c5.js:45:256)
...
(8 additional frame(s) were not displayed)
```

**Comment by catInOrbit** [comment]
## TLDR:
Clicking on stripe button somehow makes front-end passes non-existent event_id to backend causing failure. Small validation added to ensure None type event is ignore

Link to PR https://github.com/fossasia/open-event-server/pull/8947

## The Problem
Tracing log inside opev-web container it reveals that /events/<int:event_id>/stripe-authorization receives incorrect API call when Stripe radio button is clicked, for some reason a non-existing eventID **68** was passed causing a 500 Internal Server Error. Appears that stripe-authorization was called twice ?

![Image](https://github.com/fossasia/open-event-frontend/assets/49814026/c5fe3c6d-3159-485b-a589-77a27cb96277)

Video screen recordings if you need to see it in action

[**FAILURE CASE**](https://drive.google.com/file/d/1MnvRqHf86l_QQmf8hubS0_p8yUWYmjXF/view?usp=drive_link)

[**WORKING CASE AFTER VALIDATION**](https://drive.google.com/file/d/1xmmKczbFHLmrN6B2Pr4MKoOAkAaWMNzX/view?usp=drive_link)


Suspecting something may be wrong with Frontend asynchronous function here which makes the call twice

/app/controllers/orders/pending.js

```javascript
@action
  async getPaymentElement() {
    this.set('isLoading', true);
    const { order } = this.model;
    await order.save();
```

_One confusing thing is that test.event.yay.com still works without these issue ?_

## The Solution
**A working solution** for now is to have a small validation logic inside User model to not process Event data when non-existing event id was received. However I'd want to check  the full mapping of Marshmallow relationship schema to see how many API endpoints uses UserSchema. But this will cost some time

#### Frondend will have to recheck **async getPaymentElement()** to see if there's any duplicate call
